pipeline/subtraction.py

```diff
@@ -225,49 +236,335 @@ def _subtract_zogy(self,
+                newstars = newstars.iloc[ sortdex ].reset_index()
+                refstars = refstars.iloc[ sortdex ].reset_index()
+
+            SCLogger.debug( f"_subtract_hotpants using {len(newstars)} stars for kernel determination" )
+            with open( substamp_file, "w" ) as ofp:
+                for i in range(len(newstars)):
+                    ofp.write( f'{newstars.x[i]} {newstars.y[i]}\n' )
+
+            com = [ 'hotpants',
+                    '-inim', newim,
+                    '-tmplim', refim,
+                    '-outim', subim,
+                    '-tni', refnoise,
+                    '-ini', newnoise,
+                    '-tmi', refflags,
+                    '-imi', newflags,
+                    '-oni', subnoise,
+                    '-omi', subflags,
+                    '-oci', refconv,
+                    '-hki',
+                    '-n', 'i',
+                    '-c', 't',
+                    '-tl', str(tl),
+                    '-tu', str(tu),
+                    '-il', str(il),
+                    '-iu', str(iu),
+                    '-r', str(kernelrad),
+                    '-rss', str(rss),
+                    '-ssf', substamp_file,
+                    '-v', "0",
+                    '-nrx', "1",
+                    '-nry', "1",
+                    '-ko', "1",    # Maybe make this configurable?  Order of kernel spatial variation
+                    '-bgo', "1"    # Order of background variation.  Since we are not doing bgsubbed news, this matters
+                    ]
+            com.extend( gaussparam )
+            SCLogger.debug( f"Running hotpants with command: {com}" )
+            res = subprocess.run( com, shell=False, timeout=600, check=True, capture_output=True )
+            SCLogger.debug( f"...done running hotpants." )
+            if res.returncode != 0:
+                raise SubprocessFailure( res )
+
+            # Read the results and return
+            retval = {}
+            with fits.open( subim ) as hdul:
+                retval['outimhdr'] = hdul[0].header
+                retval['outim'] = hdul[0].data
+            with fits.open( subflags ) as hdul:
+                retval['outfl'] = hdul[0].data
+            with fits.open( subnoise ) as hdul:
+                noise = hdul[0].data
+
+            # Have to turn noise back into weight
+            wt = np.empty_like( noise, dtype=np.float32 )
+            wbad = ( noise <= 0 )
+            wgood = ( ~wbad )
+            wt[ wgood ] = 1. / ( noise[wgood] **2 )
+            wt[ wbad ] = 0.
+            wt[ retval['outfl'] != 0 ] = 0.
+            retval['outwt'] = wt
+
+            SCLogger.debug( f"_subtract_hotpants returning." )
+            return retval
+
+        # except Exception as ex:
+        #     # This is just here for debugging so I can see
+        #     #   what the exception was before it jumps to
+        #     #   the finally block.
+        #     import pdb; pdb.set_trace()
+        #     raise
```

Comment:
double checking if you wanted this comment block to stay around for future debugging or not

Reply:
Yeah, let's leave it here.  Future code archeologists decades or centuries will gain insight into our thought processes, and we don't want to deprive them of that.